topology/graph/graph.go

```diff
@@ -614,15 +612,22 @@ func (n *Node) Decode(i interface{}) error {
 	return n.graphElement.Decode(i)
 }
 
+func (e *Edge) MatchMetadata(f GraphElementMatcher) bool {
+	if f == nil {
```

Comment:
GraphElementMatcher is an interface thus you can't check equivalence to nil this why either:
- add to GraphElementMatcher a method IsNil()
- use a switch type
- user reflection

Reply:
@hunchback This is a reference to an Interface. It can be nil